### DISCUSSION TITLE
every user seems to be signed up - after auth via oidc

### DISCUSSION THREAD

**Discussion by 1mfaasj** [outline/outline#7778] [Self Hosting]
Hi
I've got one quick question.  How can I ensure that new users (if they auth via OIDC) are automatically registered and can see everything? is there an option to NOT allow new users?

Because I now have setup OIDC with Google, it works for every gmail account I try to use...

Thanks
Kind regards

**Comment by tommoor** [comment]
If using OIDC, then the OIDC provider is where you primarily manage access – the idea is that anyone with access to the OIDC provider can also access Outline. If using Google, you should use the Google auth method that exists separately for this reason.

There is also an option under **Settings -> Security** to restrict access to specific domains, or only to invited users.

**Reply by 1mfaasj** [reply]
Thanks for the reply. Because the Google auth method requires Google workspace, I thought give OIDC a try with Google. That works but it gives access for everyone that logs in with a Google account haha. 

In the security settings, you mean one of these options?
Allow users to send invites: Its already turned off, but when I login with a random Google account, it's still logs me in.
Allowed domain: It's only about domains, not to allow only specific users unfortunately.

hm I've looked at Google cloud console where I created the project, but there is no option to restrict users either.

So I guess I need a workspace license, or allow every Google account to access the app, or disable OIDC and use an own oauth2 service to put in front Outline...or is there maybe a way to create a validation in Outline?
